tools/admin-console/templates/healthauthority.html

```diff
@@ -0,0 +1,129 @@
+        {{else if .Thru.IsZero}}
+          <form method="POST" action="/healthauthoritykey/{{$.ha.ID}}/revoke/{{.Version}}"><button type="submit" class="btn btn-outline-danger">Revoke</button></form>
+        {{else}}
+          <form method="POST" action="/healthauthoritykey/{{$.ha.ID}}/reinstate/{{.Version}}"><button type="submit" class="btn btn-outline-warning">Clear Expiry Time</button></form>
+        {{end}}
+      </li></ul>
+    </li>
+  {{end}}
+</ul>
+
+
+{{else}}
+<div class="alert alert-warning" role="alert">
+No keys currently configured.
+</div>
+{{end}}
+
+<h4>Create new Key for <em>{{.ha.Issuer}}</em></h4>
+<form method="POST" action="/healthauthoritykey/{{.ha.ID}}/create/new" class="form-horizontal">
+  <div class="form-group row">
+    <label class="control-label col-sm-3" for="Version">Version:</label>
+    <div class="col-sm-6">
+      <input type="text" id="Version" name="Version" size="50" value="{{.hak.Version}}">
+      <small id="VersionHelpBlock" class="form-text text-muted">Shared opaque string
+      between health authority server and this server.</small>
+    </div>
+  </div>
+  <div class="form-group row">
+    <label class="control-label col-sm-3" for="PublicKeyPEM">Public Key PEM Format:</label>
+    <div class="col-sm-6">
+      <textarea class="form-control" name="PublicKeyPEM" id="PublicKeyPEM" rows="4" size="80">{{.hak.PublicKeyPEM}}</textarea>
+      <small id="RegionsHelpBlock" class="form-text text-muted">ECDSA p256 Public Key in
+        <a href="https://en.wikipedia.org/wiki/Privacy-Enhanced_Mail" target="_blank">PEM</a> format.</small>
+    </div>
+  </div>
+  <div class="form-group row">
+    <label class="control-label col-sm-3" for="enddate">Start Date/Time:</label>
+    <div class="col-sm-6">
+      <input type="date" id="fromdate" name="fromdate" value="{{.hak.FromHTMLDate}}" min="2020-05-01" max="2029-12-21-" />
+      <input type="time" id="fromtime" name="fromtime" value="{{.hak.FromHTMLTime}}" />
+      <a href="https://www.timeanddate.com/worldclock/timezone/utc" target="_blank">UTC</a>
+
+      <small id="enddateHelpBlock" class="form-text text-muted">Adjust if future
+        effective time is needed.</small>
+    </div>
+  </div>
+  <div class="form-group row">
+    <label class="control-label col-sm-3" for="enddate">End Date/Time:</label>
+    <div class="col-sm-6">
+      <input type="date" id="thrudate" name="thrudate" value="{{.hak.ThruHTMLDate}}" min="2020-05-01" max="2029-12-21-" />
+      <input type="time" id="thrutime" name="thrutime" value="{{.hak.ThruHTMLTime}}" />
+      <a href="https://www.timeanddate.com/worldclock/timezone/utc" target="_blank">UTC</a>
+
+      <small id="enddateHelpBlock" class="form-text text-muted">Leave blank
+      if this is the current key version.</small>
+    </div>
+  </div>
+  <hr/>
+  <button type="submit" class="btn btn-outline-primary" name="TODO" value="save">Create Key</button>
```

Comment:
TODO

Reply:
(Everywhere)

Reply:
that was the actual form name, but it's bad. changed.